python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -753,6 +828,19 @@ def _pack_value(
             )
         enum_serializer = whitelist_map.get_enum_entry(klass_name)
         return {"__enum__": enum_serializer.pack(val, whitelist_map, descent_path)}
+    if (
+        (isinstance(val, tuple) and hasattr(val, "_fields"))
+        or (is_dataclass(val) and not isinstance(val, type))
+        or isinstance(val, pydantic.BaseModel)
```

Comment:
did some benchmarking,  hoisting the `isinstance(val, Enum)` check above since that is cheaper prevents these noticeably more expensive checks from firing too often on existing large serdes payloads.